vivisect/vamp/__init__.py

```diff
@@ -42,10 +42,13 @@ def genSigAndMask(vw, funcva):
     """
 
     fsize = 0
-
-    # Figgure out the size of the first linear chunk
+    if startva == None:
+        startva = funcva
+    # Figure out the size of the first linear chunk
     # in this function...
-    cb = vw.getCodeBlock(funcva)
+    cb = vw.getCodeBlock(startva)
+    if cb[2] != funcva:
+        raise Exception("startva not in given func")
```

Comment:
this is a trap.  Viv currently only tracks one possible function for each codeblock, although in reality, this is not always so.  if that codeblock is the function start, this isn't a problem.  however, if it's not, this becomes a problem.
iinm, currently the only accurate way to access the function's codeblocks:

vw.getFunctionBlocks(funcva)

something like
if cb[CB_VA] not in [cbva for cbva,cbsz,cbfva in vw.getFunctionBlocks(funcva)]:

or something more elegant... but equally as accurate.

Reply:
Good catch. I'll change that around.